docs/advanced-features/middleware.md

```diff
@@ -218,6 +218,44 @@ export function middleware(request: NextRequest) {
 
 > **Note:** Avoid setting large headers as it might cause [431 Request Header Fields Too Large](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/431) error depending on your backend web server configuration.
 
+## Producing a Response
+
+You can respond to middleware directly by returning a `NextResponse` (responding from middleware is available since Next.js v13.0.0).
+
+To enable middleware responses, update `next.config.js`:
+
+```js:next.config.js
+module.exports = {
+  experimental: {
+    allowMiddlewareResponseBody: true
+  },
+};
+```
+
+Once enabled, you can provide a response from middleware using the `NextResponse` API:
```

Comment:
This can also be a standard response e.g. a `fetch` `Response`.

Reply:
updated